user.py

```diff
@@ -0,0 +1,249 @@
+        """Add a user to the database"""
+        user_data = self.get_user(dictionary=True)
+        user_data.pop('id')  # ensure key assignment is handled by database
+        cr = db.cursor()
+        # Insert new user
+        cr.execute(f"INSERT INTO user ({', '.join(user_data.keys())}) VALUES {tuple(user_data.values())}")
+        if self.commit_to_db:
+            db.commit()
+        cr.close()
+
+    def __eq__(self, user):
+        """equate a user to another user by ids as they are unique"""
+        return self.id == user.id
+
+    def __hash__(self):
+        """implemented in order to use users as keys in dict"""
+        return hash(self.get_user(hide_password=True))
+
+    def __repr__(self):
+        """string representation of a user which displays username for shorthand"""
+        return f'User({self.username})'
+
+
+if __name__ == '__main__':
+    import mysql.connector
+
+    db = mysql.connector.connect(
+        host='192.168.1.53',
+        user='root',
+        passwd='Binstagram_341',
+        database='binstagram'
+    )
+
+    # add users to db from csv
+    # with open('data/users.csv', 'r') as f:
+    #     for user in f.read().splitlines()[1:]:
+    #         User.addUser(db, User(*user.split(',')))
+
+    # add images to db from directory /static/user_images
+    # cr = db.cursor()
+    # for i in range(1, 258):
+    #     data = (get_image(f'static/user_images/img({i}).jpg'),)
+    #     cr.execute("INSERT INTO image (data) VALUES (%s)", data)
+    #     db.commit()
+    # cr.close()
+
+    # retrieve all images from db, output to test directory
+    # cr = db.cursor(dictionary=True)
+    # cr.execute("SELECT data FROM image")
+    # data = cr.fetchall()
+    # cr.close()
+    # images = list(map(lambda x: x['data'], data))
+    # for i in range(len(images)):
+    #     with open(f'test/img({i+1}).jpg', 'wb') as f:
+    #         f.write(images[i])
+
+    # add follower relationships to db
+    # cr = db.cursor()
+    # with open('data/followers.csv', 'r') as f:
+    #     follower = f.read().splitlines()[1:]
+    # for i in follower:
+    #     users = i.split(',')
+    #     user_id = User.getByUsername(db, users[0]).id
+    #     # print(users[0], user_id)
+    #     for following in users[1:]:
+    #         following_id = User.getByUsername(db, following).id
+    #         # print(user_id, following_id, sep=' -> ')
+    #         cr.execute(f"INSERT INTO follower (user_id, following_id) VALUES ('{user_id}', '{following_id}')")
+    #         db.commit()
+    # cr.close()
+
+    # print(User.get_usernames(db))
+    user = User.get_by_username(db, 'Ablion73')
+    # print(user.get_user(dictionary=True, hide_password=True))
+    user_by_id = User.get_by_id(db, 1)
+    assert user == user_by_id, 'ids are different'
+    # followable = user.get_followable(db)
+    # print(user.id)
+    # print(followable)
+    # print(user.get_likes(db))
+    print(user.get_following_post_images(db))
```

Comment:
Are these supposed to not be commented (lines 228 - 236)? Are they for testing or for regular functionality?